### ISSUE TITLE
[Feature/Improvement] change "str _unit" to netId

### ISSUE THREAD

**Issue by BlackFisch** [AsYetUntitled/Framework#613]
### Requested Change
Currently, when units are involved something like `str _unit` or `str player` is added to the listbox. Instead, `netId _unit` should be used instead (object can be retrieved again via `objectFromNetId "netId_here"` and would be way better than `call compile _str` everytime

-> this is more just a reminder for myself than anything else so I don't forget :D Will do the PR when #611 and #612 are merged (to not fuck up everything)

**Comment by Azeh** [comment]
Can do `missionNamespace getVariable _str` instead of `call compile _str`.

**Comment by BlackFisch** [comment]
not really. Doesn't even matter, its not intended to use the string representation of an object to store it and compile it back... That's what netId's are for :)

**Comment by Azeh** [comment]
You were talking about listboxes and units.
Something like
```
private _unit = lbData [idc, lbCurSel idc];
missionNamespace getVariable _unit;
```
works the same way as 
```
private _unit = lbData [idc, lbCurSel idc];
call compile _unit;
```

**Comment by BlackFisch** [comment]
The problem is with the contents. Like [here](https://github.com/AsYetUntitled/Framework/blob/v5.X.X/Altis_Life.Altis/core/pmenu/fn_p_updateMenu.sqf#L34) most of the time in the framework the data set is something like `str _unit` -> that cannot be resolved to an object with `missionNamespace getVariable` (as shown in the screenshot above) but has to be done with `call compile _str` which is memory intensive and clutters the memory -> netId is the solution to this "problem"

**Comment by Azeh** [comment]
`something like str _unit -> that cannot be resolved to an object with missionNamespace getVariable`
It can.

**Comment by BlackFisch** [comment]
I tried it (see screenshot) and it didn't work. You can prove me otherwise of course, just add me some working code for proof 

still. netId is the way to go. It is not intended use to str and object to refer to it.

**Comment by Azeh** [comment]
![](https://i.gyazo.com/85bc3e1a72d6425bd3cb39047aee26c3.png)

![](https://i.gyazo.com/0cdfa04ad5dce9c4cddfd7c0539fa36e.png)

**Comment by BlackFisch** [comment]
This only works with named units as @DomT602 pointed out on the discord. This does not work with vehicles and such. Thats it :D
Problem is, this "method" is also used in vehicles lists (keychain etc) -> that would not work and cause problems. thats why I say netId :D
